### ISSUE TITLE
Failed to reproduce the reported results

### ISSUE THREAD

**Issue by ChaseDreamInfinity** [CVLAB-Unibo/Real-time-self-adaptive-deep-stereo#16]
Thanks to your good work, but I failed to reproduce your results on 2011_09_30_drive_0028_sync sequence. Here is the command that I used with the synthetic pretrained MADNet provided by you:

```
python Stereo_Online_Adaptation.py \
    -l ${LIST of 2011_09_30_drive_0028_sync  sequence} \
    -o ${OUTPUT} \
    --weights ${WEIGHTS_FOLDER}/MADNet/synthetic/weights.ckpt \
    --modelName MADNet \
    --blockConfig config/MadNet_full.json \
    --mode ${MODE} \
    --sampleMode PROBABILITY \
    --logDispStep 1
```

According to your [video](https://www.youtube.com/watch?v=7SjyzDxmCY4), on frame 1998 (corresponding to the actual frame 0000002003.png), 
When ${MODE} = MAD, then EPE = 0.30
When ${MODE} = FULL, then EPE = 0.16

However, here are what I get
When setting ${MODE} = MAD, then EPE = 4.29
When setting ${MODE} = FULL, then EPE = 2.62
which are unsatisfactory. (I do did center cropping on the ground truth before comparison. BTW, I used the following code to convert the depth on 2011_09_30 to disparity)

```
baseline = 0.54
focal_len = 7.070912e+02
disparity = focal_len * baseline / depth * 256
disparity[depth==0] = 0
disparity = disparity.astype('uint16')
```

Here are some printed intermediate results when I setting ${MODE}=FULL,
Step:   0	bad3:0.54	EPE:4.62	SSIM:0.14	f/b time:0.085308	Missing time:0:07:20.783953
Step: 100	bad3:0.46	EPE:4.11	SSIM:0.11	f/b time:0.083690	Missing time:0:07:04.058810
Step: 200	bad3:0.48	EPE:3.74	SSIM:0.14	f/b time:0.085709	Missing time:0:07:05.718425
Step: 300	bad3:0.30	EPE:2.82	SSIM:0.11	f/b time:0.078649	Missing time:0:06:22.782704
Step: 400	bad3:0.30	EPE:2.46	SSIM:0.11	f/b time:0.076860	Missing time:0:06:06.390148
Step: 500	bad3:0.28	EPE:2.62	SSIM:0.11	f/b time:0.077608	Missing time:0:06:02.194405
Step: 600	bad3:0.47	EPE:5.23	SSIM:0.16	f/b time:0.078980	Missing time:0:06:00.702607
Step: 700	bad3:0.25	EPE:2.23	SSIM:0.17	f/b time:0.079465	Missing time:0:05:54.969689
Step: 800	bad3:0.41	EPE:4.13	SSIM:0.09	f/b time:0.078717	Missing time:0:05:43.755819
Step: 900	bad3:0.31	EPE:2.69	SSIM:0.10	f/b time:0.077609	Missing time:0:05:31.156342
Step:1000	bad3:0.25	EPE:2.25	SSIM:0.17	f/b time:0.078229	Missing time:0:05:25.981169
Step:1100	bad3:0.14	EPE:1.65	SSIM:0.12	f/b time:0.078017	Missing time:0:05:17.293466
Step:1200	bad3:0.28	EPE:2.58	SSIM:0.13	f/b time:0.077899	Missing time:0:05:09.024494
Step:1300	bad3:0.22	EPE:2.09	SSIM:0.13	f/b time:0.080689	Missing time:0:05:12.023805
Step:1400	bad3:0.21	EPE:2.05	SSIM:0.12	f/b time:0.083007	Missing time:0:05:12.688912
Step:1500	bad3:0.14	EPE:1.70	SSIM:0.14	f/b time:0.083191	Missing time:0:05:05.061603
Step:1600	bad3:0.21	EPE:2.20	SSIM:0.11	f/b time:0.081222	Missing time:0:04:49.718330
Step:1700	bad3:0.38	EPE:3.53	SSIM:0.13	f/b time:0.082633	Missing time:0:04:46.489179
Step:1800	bad3:0.14	EPE:1.63	SSIM:0.10	f/b time:0.079664	Missing time:0:04:28.229809
Step:1900	bad3:0.35	EPE:2.52	SSIM:0.13	f/b time:0.079072	Missing time:0:04:18.326887
Step:2000	bad3:0.27	EPE:2.33	SSIM:0.09	f/b time:0.080154	Missing time:0:04:13.848781

Could you please advise?

**Comment by AlessioTonioni** [comment]
The video was done a lot of time ago and I'm not sure anymore of the exact hyperparameters. But either way, your results seem way worse than what I get. 
Here you can find the command line and output of this code on my pc on the same sequence
https://drive.google.com/open?id=1m2U0S_5P3ubRAoWool8L_RgnLr9XUpqv

**Comment by ChaseDreamInfinity** [comment]
Thanks for providing the intermediate results. To align with your set up, I changed the learning rate to 0.001 instead of using the default 0.0001, it become a little better, but still not satisfied.

Mine:
Iteration,Time,EPE,bad3
0,0.0,4.624141216278076,0.542962372303009
1,0.0502320025416749,6.039391040802002,0.5856813788414001
2,0.1004640050833498,5.362223148345947,0.5551557540893555

Yours:
Iteration,Time,EPE,bad3
0,0.0,3.1957602500915527,0.25764191150665283
1,0.03698867853591768,3.4075660705566406,0.2395375818014145
2,0.07397735707183536,2.54512357711792,0.22660379111766815

As you can see, even the results of the 0-th iteration is different. So I guess
1. You've used a better pre-trained model
2. The adaptive scheme actually fails to adapt with different pre-trained models.

**Comment by AlessioTonioni** [comment]
No the weights are exactly the same that we provide on the main web page of the repository. 
Which depth map have you used to get the disparity maps? The raw lidar scans or the one available here: http://www.cvlibs.net/datasets/kitti/eval_depth.php?benchmark=depth_prediction?

**Comment by ChaseDreamInfinity** [comment]
data_depth_annotated.zip

**Comment by mattpoggi** [comment]
Hi @ChaseDreamInfinity ,
by casting disparity to uint16 you are losing subpixel accuracy, you should keep it float32. Anyway, I don't think this is the only cause behind your numbers. 
I assume you are opening depth maps as uint16, right?
